src/cpp/session/modules/SessionFiles.cpp

```diff
@@ -1391,26 +1391,35 @@ class ListFilesAcceptAll
 class ListFilesAcceptMatching
 {
 public:
-#ifdef BOOST_WINDOWS_API
-   typedef boost::wregex regex_type;
-   typedef std::wstring string_type;
-#else
-   typedef boost::regex regex_type;
-   typedef std::string string_type;
-#endif
    
-   ListFilesAcceptMatching(const string_type& pattern, int flags)
-      : pattern_(pattern, flags)
+   ListFilesAcceptMatching(SEXP patternSEXP, SEXP ignoreCaseSEXP)
    {
+      // construct our call up-front, so we can re-use it without
+      // paying the cost to rebuild this call on every invocation
+      callSEXP_ = Rf_lang4(
+               Rf_install("grepl"),    // function
+               patternSEXP,            // pattern
+               R_NilValue,             // x (filled in below)
+               ignoreCaseSEXP);        // ignore.case
+
+      preserver_.add(callSEXP_);
    }
    
    bool operator()(const boost::filesystem::path& path)
    {
-      return boost::regex_search(path.native(), pattern_);
+      // construct call
+      r::sexp::Protect protect;
+      SEXP xSEXP = r::sexp::create(path.string(), &protect);
+      SETCADDR(callSEXP_, xSEXP);
```

Comment:
does this just set xSEXP to the second argument because that one happens to be R_NilValue?

Reply:
That's right. The R call object is basically like a list of elements ["grepl", <pattern>, <x>, <ignore.case>]; to set the 3rd element in that list (the second argument), we use `SETCADDR()`.